manifests/server/passwd.pp

```diff
@@ -1,6 +1,11 @@
 # @api private
 class postgresql::server::passwd {
-  $postgres_password = $postgresql::server::postgres_password
+  $postgres_password = if $postgresql::server::postgres_password =~ Sensitive {
```

Comment:
Just one comment, and I apologize for it being at the very last second, but the unwrap shown here is not needed from what I understand.
According to the official documentation on the type, the sensitive data type only effects the logs and reports, when it comes to the actual changes being applied it has no effect. Unwrap is only used in situations where you want the value to be output into a specific report or log.
```
Sensitive types in the Puppet language are strings marked as sensitive.
The value is displayed in plain text in the catalog and manifest, but is redacted from logs and reports.
Because the value is maintained as plain text, use it only as an aid to ensure that sensitive values are not inadvertently disclosed.
```
This is taken from the documentation for the type shown here: https://puppet.com/docs/puppet/7/lang_data_sensitive.html

Again, apologies for not mentioning this earlier but I only just became aware of this after utilising the Sensitive type in a pr of my own.

Reply:
Gonna resolve this as the other use's on unwrap are valid and seem necessary for the code to work as they involve manipulation of the value.
Sorry for the bother.

Reply:
> but the unwrap shown here is not needed from what I understand.

My main Goal is, to make Component-Modules like `postgresql` to accept `Sensitive`, because an Author of Profile-Modules should not bother with `unwrap`.  Component-Modules "just have to work".  My `if`-Clause is cheap Means to achive this.
But for the Internals of the Module `postgresql` you are right, in the Way, that `unwrap` should happen in the least possible Situation.  This means: not the Class `postgresql::server::passwd` should take Care of it, but instead the following Function `postgresql::postgresql_escape` and the `exec`.